Public/Src/Cache/VerticalStore/Interfaces/CacheFactory.cs

```diff
@@ -165,6 +166,69 @@ public static string Serialize(this ICacheConfigData cacheData)
             return await factoryObject.InitializeCacheAsync(cacheData, activityId);
         }
 
+        /// <summary>
+        /// Validates that the config data is valid.
+        /// It loads cache factory assemblies and calls the right validation method.
+        /// </summary>
+        public static IEnumerable<Failure> ValidateConfig(ICacheConfigData cacheData)
+        {
+            Contract.Requires(cacheData != null);
+
+            var failures = new List<Failure>();
+            object value;
+            if (!cacheData.TryGetValue(DictionaryKeyFactoryAssemblyName, out value))
+            {
+                failures.Add(new IncorrectJsonConfigDataFailure("Cache factory Assembly name is required, but it was not specified"));
+            }
+
+            string assembly = value.ToString();
+
+            if (!cacheData.TryGetValue(DictionaryKeyFactoryTypeName, out value))
+            {
+                failures.Add(new IncorrectJsonConfigDataFailure("Cache factory Type name is required, but it was not specified"));
+            }
+
+            if (failures.Any())
+            {
+                return failures;
+            }
+
+            string type = value.ToString();
+
+            ICacheFactory factoryObject;
+            Exception instantiationException = null;
+            try
+            {
+                Assembly assemblyFile = Assembly.Load(assembly);
+                Type myType = assemblyFile.GetType(type);
+                if (myType == null)
+                {
+                    throw new ArgumentException($"Typename {type} could not be found in {assembly}");
+                }
+                factoryObject = Activator.CreateInstance(myType) as ICacheFactory;
+            }
+            catch (Exception ex)
+            {
+                // We failed to produce an instance of the specified type. We will return a Failure from the next if statement
+                factoryObject = null;
+                instantiationException = ex;
+            }
+
+            // Make sure that the cache factory is an ICacheFactory.

```

Comment:
>// Make sure that the cache factory is an ICacheFactory. [](start = 12, length = 56)

`// Build error message for failed ICacheFactory construction` ? We aren't making sure of anything here - the operation already failed.